_posts/2023/03/2023-03-13-probing.md

```diff
@@ -0,0 +1,178 @@
+---
+layout: post
+date: 2023-03-13
+title: "Probing"
+author: volodya-lombrozo
+---
+
+## Overview
+
+The EO language has a convenient feature that enables the use of objects
+without the need to specify their source, location or origin. A developer can
+simply print the name of the object and that is all that is required. The
+compiler will automatically locate all the requested objects through the
+implementation of a new probing feature, which was recently added. In this short
+blog, we will attempt to explain how this feature works. Let's take a look at
+the following code snippet:
+
+```eo
+[] > app
+  org.eolang.io.stdout > @
+    org.eolang.txt.sprintf
+      "Magic number is %d"
+      4815162342
+```
+
+As you can see from the code, the purpose of this program is to print the magic
+number to the standard output. However, what is `org.eolang.io.stdout` and
+`org.eolang.io.stdout` here? They all are objects, right.
+Moreover, `org`, `org.eolang` , `org.eolang.io` are objects too.
+
+But where are they? Is it `org.eolang.io` object that contains `stdout`
+attribute, or `org.eolang.io.stdout` is the top-level object itself?
+In other words, the compiler needs to distinguish between objects that
+are part of other objects (attributes) and those that are top-level objects.
+For instance, `org.eolang.io.stdout` is a top-level object, whereas
+`org.eolang.io.stdout.length` is an object-attribute of `org.eolang.io.stdout`.
+
+Hence, the compiler must have the ability to locate all of these objects and
+make the process transparent for the developer. This process is called
+_probing_.
+
+## Implementation
+
+During the compilation of the application from the example above, the compiler
+will create a list of all possible objects that the application might require
+and add them to the `meta` section. This list ensures that the compiler is aware
+of all the necessary objects and will try to locate them as needed:
```

Comment:
@volodya-lombrozo "all the necessary objects" -> "all object candidates"

Reply:
@mximp Done, thanks!